### PR TITLE
Add a configurable analytics reducer. Track GoogleAnalytics

### PR DESCRIPTION
**Pull request by nicosantangelo**
# README

1. I tried to keep the code generic and extractable so we can reuse it later (and move it into `commons` !). I didn't yet abstract it away into a library to battle test it first
2. This is reporting `dev events` without any kind of check (https://github.com/decentraland/auction/compare/analytics?expand=1#diff-988ccd9de4f73ecd0f56726a68e08a2eR36). We could add something like `if (env.isProduction()) configure()`, but as I said [here](https://github.com/decentraland/auction/pull/30), I'm afraid this could lead to problems with `test/beta` deployments. SO I advice we filter `localhost` events **from the analytics platform itself**
3. The code on `index.html` can easily be extracted into `<GoogleAnalytics />` and injected dynamically but I'm not sure if it's worthwhile. On one hand we stop having to add extra code outside `React`, on the other, we **have** to make sure the component is always mounted which right now amounts to adding a new `<Page>` component to wrap every route. What do you think? (dear reviewer)

That's it!

### CODE REVIEW

File: webapp/src/containers/GoogleAnalyticsContainer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { stateData } from '../lib/propTypes'
+
+class GoogleAnalyticsContainer extends React.Component {
+  static propTypes = {
+    addressState: stateData(PropTypes.object).isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    this.configurated = false
```
Comment (abarmat) [inline webapp/src/containers/GoogleAnalyticsContainer.js:15]: When is this variable changing to true?

**Review by abarmat — COMMENTED** [review]
1) Nice!
2) Agree
3) I think it's good to have it packaged in a way we can easily use in other places. But as you said not sure it's worthwhile.

File: webapp/src/containers/GoogleAnalyticsContainer.js (hunk 1 above)
Comment (nicosantangelo) [reply webapp/src/containers/GoogleAnalyticsContainer.js:15]: Nice catch! it should happen on the `config`  method below. Fixing!

Comment (abarmat) [reply webapp/src/containers/GoogleAnalyticsContainer.js:15]: Superb.

**Pull request merged**